PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:12 + add:0, change matches:
-    _inherit = ["account.move"]
-    _inherit = ['account.move.line']
-    _inherit = ['hr.employee']
-    _inherit = ['account.analytic.line']
-    _inherit = ['product.product']
-    _inherit = ['product.template']
-    _inherit = ['project.project']
-    _inherit = ["project.task"]
-    _inherit = ['project.update']
-    _inherit = ['res.config.settings']
-    _inherit = ['sale.order']
-    _inherit = ["sale.order.line"]

Total Changes: 25

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale_timesheet/models/account_move.py
+++ b/addons/sale_timesheet/models/account_move.py
@@ -7,7 +7,7 @@ from odoo.osv import expression
 
 
 class AccountMove(models.Model):
-    _inherit = ["account.move"]
+    _inherit = "account.move"
 
     timesheet_ids = fields.One2many('account.analytic.line', 'timesheet_invoice_id', string='Timesheets', readonly=True, copy=False, export_string_translation=False)
     timesheet_count = fields.Integer("Number of timesheets", compute='_compute_timesheet_count', compute_sudo=True, export_string_translation=False)

--- a/addons/sale_timesheet/models/account_move_line.py
+++ b/addons/sale_timesheet/models/account_move_line.py
@@ -6,7 +6,7 @@ from odoo import api, models
 
 
 class AccountMoveLine(models.Model):
-    _inherit = ['account.move.line']
+    _inherit = 'account.move.line'
 
     @api.model
     def _timesheet_domain_get_invoiced_lines(self, sale_line_delivery):

--- a/addons/sale_timesheet/models/hr_employee.py
+++ b/addons/sale_timesheet/models/hr_employee.py
@@ -4,7 +4,7 @@ from odoo import api, models
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     @api.model
     def default_get(self, fields):

--- a/addons/sale_timesheet/models/hr_timesheet.py
+++ b/addons/sale_timesheet/models/hr_timesheet.py
@@ -21,7 +21,7 @@ TIMESHEET_INVOICE_TYPES = [
 
 
 class AccountAnalyticLine(models.Model):
-    _inherit = ['account.analytic.line']
+    _inherit = 'account.analytic.line'
 
     def _domain_so_line(self):
         domain = expression.AND([

--- a/addons/sale_timesheet/models/product_product.py
+++ b/addons/sale_timesheet/models/product_product.py
@@ -7,7 +7,7 @@ from odoo.exceptions import ValidationError
 
 
 class ProductProduct(models.Model):
-    _inherit = ['product.product']
+    _inherit = 'product.product'
 
     @tools.ormcache()
     def _get_default_uom_id(self):

--- a/addons/sale_timesheet/models/product_template.py
+++ b/addons/sale_timesheet/models/product_template.py
@@ -7,7 +7,7 @@ from odoo.exceptions import ValidationError
 
 
 class ProductTemplate(models.Model):
-    _inherit = ['product.template']
+    _inherit = 'product.template'
 
     def _selection_service_policy(self):
         service_policies = super()._selection_service_policy()

--- a/addons/sale_timesheet/models/project_project.py
+++ b/addons/sale_timesheet/models/project_project.py
@@ -10,7 +10,7 @@ from odoo.tools.translate import _
 
 
 class ProjectProject(models.Model):
-    _inherit = ['project.project']
+    _inherit = 'project.project'
 
     @api.model
     def default_get(self, fields):

--- a/addons/sale_timesheet/models/project_sale_line_employee_map.py
+++ b/addons/sale_timesheet/models/project_sale_line_employee_map.py
@@ -6,6 +6,7 @@ from odoo.tools.misc import unquote
 
 
 class ProjectSaleLineEmployeeMap(models.Model):
+    _name = 'project.sale.line.employee.map'
     _description = 'Project Sales line, employee mapping'
 
     def _domain_sale_line_id(self):

--- a/addons/sale_timesheet/models/project_task.py
+++ b/addons/sale_timesheet/models/project_task.py
@@ -5,7 +5,7 @@ from odoo.osv import expression
 
 
 class ProjectTask(models.Model):
-    _inherit = ["project.task"]
+    _inherit = "project.task"
 
     def _get_default_partner_id(self, project, parent):
         res = super()._get_default_partner_id(project, parent)

--- a/addons/sale_timesheet/models/project_update.py
+++ b/addons/sale_timesheet/models/project_update.py
@@ -7,7 +7,7 @@ from odoo.tools.misc import format_duration
 
 
 class ProjectUpdate(models.Model):
-    _inherit = ['project.update']
+    _inherit = 'project.update'
 
     @api.model
     def _get_template_values(self, project):

--- a/addons/sale_timesheet/models/res_config_settings.py
+++ b/addons/sale_timesheet/models/res_config_settings.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     invoice_policy = fields.Boolean(string="Invoice Policy", help="Timesheets taken when invoicing time spent")

--- a/addons/sale_timesheet/models/sale_order.py
+++ b/addons/sale_timesheet/models/sale_order.py
@@ -8,7 +8,7 @@ from odoo.tools import float_compare
 
 
 class SaleOrder(models.Model):
-    _inherit = ['sale.order']
+    _inherit = 'sale.order'
 
     timesheet_count = fields.Float(string='Timesheet activities', compute='_compute_timesheet_count', groups="hr_timesheet.group_hr_timesheet_user", export_string_translation=False)
     timesheet_encode_uom_id = fields.Many2one('uom.uom', related='company_id.timesheet_encode_uom_id', export_string_translation=False)

--- a/addons/sale_timesheet/models/sale_order_line.py
+++ b/addons/sale_timesheet/models/sale_order_line.py
@@ -6,7 +6,7 @@ from odoo.tools import format_duration
 
 
 class SaleOrderLine(models.Model):
-    _inherit = ["sale.order.line"]
+    _inherit = "sale.order.line"
 
     qty_delivered_method = fields.Selection(selection_add=[('timesheet', 'Timesheets')])
     analytic_line_ids = fields.One2many(domain=[('project_id', '=', False)])  # only analytic lines, not timesheets (since this field determine if SO line came from expense)
